src/dma.rs

```diff
@@ -39,7 +39,9 @@ impl<B, C: Channel, T> Transfer<B, C, T> {
         B: WriteBuffer + 'static,
         T: Target<C>,
     {
-        let (ptr, len) = buffer.write_buffer();
+        // NOTE(unsafe) cannot call `&mut self` methods on `buffer` because its
+        // concrete type is unknown here
```

Comment:
```suggestion
        // NOTE(unsafe) We can't (and won't) call `&mut self` methods on
        // `buffer` after this call and until the end of the transfer, because
        // its concretetype is unknown here
```

Reply:
Not sure about the wording, but we should explain this better.

Reply:
I thought it was clear, but I'm probably in too deep :) I'll update this with more words.